lib/facility.rb

```diff
@@ -1,14 +1,88 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, 
+              :address, 
+              :city,
+              :state,
+              :zip,
+              :phone, 
+              :website, 
+              :type, 
+              :services, 
+              :registered_vehicles, 
+              :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(information)
+    @name = information[:name]
+    @address = information[:address]
+    @city = information[:city]
+    @state = information[:state]
+    @zip = information[:zip]
+    @phone = information[:phone]
+    @website = information[:website]
+    @type = information[:type]    
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
+    valid_entries = ["Road Test", "Written Test", "Renew License", "Vehicle Registration"]
+    unless service.is_a?(String) && valid_entries.include?(service)
+      return "Invalid entry. Services offered are: #{valid_entries.join(", ")}"
+    end
     @services << service
   end
+  
+
+  def register_vehicle(car)
+    return "Service not available at selected location." unless services.include?("Vehicle Registration")
+    if car.electric_vehicle?
+      car.plate_type = :ev
+      car.registration_date = Date.today.strftime("%Y-%m-%d")
+      collect_cash(200) 
+      @registered_vehicles << car 
+    elsif car.antique? 
+      car.plate_type = :antique
+      car.registration_date = Date.today.strftime("%Y-%m-%d")
+      collect_cash(25)
+      @registered_vehicles << car 
+    else
+      car.plate_type = :regular
+      car.registration_date = Date.today.strftime("%Y-%m-%d")
+      collect_cash(100) 
+      @registered_vehicles << car 
+    end
+  end
```

Comment:
I like that you are returning something if a certain condition is met first, because then your code doesnt need to go through the entire conditional tree for an unnecessary reason.

Rather than setting the plate_type and the registration_date in this method, you could consider putting a method in the Vehicle class that will do those things for you. That would again, align more with SRP, and it would abstract that logic so that the Vehicle class is the only one that knows about it, and this class doesn't need to worry about it.